packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1420,6 +1420,175 @@ export class LinuxBuildImage implements IBuildImage {
+  /** PyTorch 1.4.0 GPU training image from AWS Deep Learning Containers. */
+  public static readonly DLC_PYTORCH_1_4_0_TRAINING = LinuxGpuBuildImage.DlcImage('pytorch-training',
+    '1.4.0-gpu-py36-cu101-ubuntu16.04');
+  /** PyTorch 1.4.0 GPU inference image from AWS Deep Learning Containers. */
+  public static readonly DLC_PYTORCH_1_4_0_INFERENCE = LinuxGpuBuildImage.DlcImage('pytorch-inference',
+    '1.4.0-gpu-py36-cu101-ubuntu16.04');
+  /** PyTorch 1.5.0 GPU training image from AWS Deep Learning Containers. */
+  public static readonly DLC_PYTORCH_1_5_0_TRAINING = LinuxGpuBuildImage.DlcImage('pytorch-training',
+    '1.5.0-gpu-py36-cu101-ubuntu16.04');
+  /** PyTorch 1.5.0 GPU inference image from AWS Deep Learning Containers. */
+  public static readonly DLC_PYTORCH_1_5_0_INFERENCE = LinuxGpuBuildImage.DlcImage('pytorch-inference',
+    '1.5.0-gpu-py36-cu101-ubuntu16.04');
+
+  /** MXNet 1.4.1 GPU image from AWS Deep Learning Containers. */
+  public static readonly DLC_MXNET_1_4_1 = LinuxGpuBuildImage.DlcImage('mxnet-training',
+    '1.4.1-gpu-py36-cu100-ubuntu16.04');
+  /** MXNet 1.6.0 GPU image from AWS Deep Learning Containers. */
+  public static readonly DLC_MXNET_1_6_0 = LinuxGpuBuildImage.DlcImage('mxnet-training',
+    '1.6.0-gpu-py36-cu101-ubuntu16.04');
+
+  /**
+   * @returns a Linux GPU build image from a Docker Hub image.
+   */
+  public static fromDockerRegistry(name: string, options: DockerImageOptions = {}): IBuildImage {
+    return new LinuxGpuBuildImage({
+      ...options,
+      imageId: name,
+      imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+    });
+  }
+
+  /**
+   * @returns A Linux GPU build image from an ECR repository.
+   *
+   * NOTE: if the repository is external (i.e. imported), then we won't be able to add
+   * a resource policy statement for it so CodeBuild can pull the image.
+   *
+   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/sample-ecr.html
+   *
+   * @param repository The ECR repository
+   * @param tag Image tag (default "latest")
+   */
+  public static fromEcrRepository(repository: ecr.IRepository, tag: string = 'latest'): IBuildImage {
+    return new LinuxGpuBuildImage({
+      imageId: repository.repositoryUriForTag(tag),
+      imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+      repository,
+    });
+  }
+
+  /**
+   * Uses an Docker image asset as a Linux GPU build image.
+   */
+  public static fromAsset(scope: Construct, id: string, props: DockerImageAssetProps): IBuildImage {
+    const asset = new DockerImageAsset(scope, id, props);
+    return new LinuxGpuBuildImage({
+      imageId: asset.imageUri,
+      imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+      repository: asset.repository,
+    });
+  }
+
+  /**
+   * @returns a Linux GPU image from AWS Deep Learning Containers.
+   */
+  private static DlcImage(name: string, tag: string): IBuildImage {
+    let account: number;
+    switch (Aws.REGION) {
```

Comment:
This is a nice attempt, but it cannot work. When this code executes, the region is still not resolved (it's only resolved at deploy time). So, this will always go to the `default` clause.

Reply:
I've modified this to use only the default account number and added a message to the documentation indicating that the special regions are not supported.